apps/test/unit/JSInterpreterTest.js

```diff
@@ -274,6 +274,312 @@ myCallback("this message is coming from inside the interpreter");
 
   });
 
+  let aceEditor;
+  function setupFakeAce() {
+    let oldAce;
+    beforeEach(() => {
+      oldAce = window.ace;
+      window.ace = {
+        require: sinon.stub().returns({Range: sinon.stub()}),
+      };
+      const breakpoints = [];
+      const aceSession = {
+        addMarker: sinon.spy(),
+        getBreakpoints: sinon.stub().returns(breakpoints),
+      };
+      aceEditor = {
+        isRowFullyVisible: () => true,
+        getSession: () => aceSession,
+        getSelection: () => ({
+          getRange: () => ({
+            start: {},
+            end: {},
+          }),
+          clearSelection: () => {},
+        })
+      };
+    });
+    afterEach(() => {
+      window.ace = oldAce;
+    });
+  }
+
+  describe("executeInterpreter method", () => {
+    let onPauseObserver;
+    setupFakeAce();
+    beforeEach(() => {
+      jsInterpreter = new JSInterpreter({
+        studioApp: {
+          hideSource: false,
+          editor: {
+            aceEditor: aceEditor,
+          },
+        },
+      });
+      onPauseObserver = sinon.spy();
+      jsInterpreter.onPause.register(onPauseObserver);
+      sinon.spy(jsInterpreter, 'handleError');
+    });
+
+    function getCurrentLine() {
+      const interpreter = jsInterpreter.interpreter;
+      const interpreterValue = interpreter.getProperty(interpreter.globalScope, 'currentLine');
+      if (interpreterValue === interpreter.UNDEFINED) {
+        return undefined;
+      }
+      return interpreterValue.toNumber();
+    }
```

Comment:
Seems a little like this method should get moved out of test code and onto the jsinterpreter prototype.